helm/cluster-azure/templates/_machine_pools.tpl

```diff
@@ -0,0 +1,81 @@
+{{- define "machine-pools" }}
+{{ range .Values.machinePools }}
+apiVersion: cluster.x-k8s.io/v1beta1
+kind: MachinePool
+metadata:
+  annotations:
+    machine-pool.giantswarm.io/name: {{ include "resource.default.name" $ }}-{{ .name }}
+  labels:
+    giantswarm.io/machine-pool: {{ include "resource.default.name" $ }}-{{ .name }}
+    {{- include "labels.common" $ | nindent 4 }}
+  name: {{ include "resource.default.name" $ }}-{{ .name }}
+  namespace: {{ $.Release.Namespace }}
+spec:
+  clusterName: {{ include "resource.default.name" $ }}
+  replicas: {{ .minSize }}
+  template:
+    spec:
+      bootstrap:
+        configRef:
+          apiVersion: bootstrap.cluster.x-k8s.io/v1beta1
+          kind: KubeadmConfig
+          name: {{ include "resource.default.name" $ }}-{{ .name }}
```

Comment:
we will need to test if we can change this for a running machinepool ( to update whatever field ) or we need a templated name with hash

Reply:
fixed (also interesting: the webhook doesn't catch this, but the controller itself is complaining about this).